sentry-core/src/main/java/io/sentry/core/Scope.java

```diff
@@ -332,7 +353,7 @@ public void removeExtra(@NotNull String key) {
 
     final Map<String, Object> extraRef = extra;
 
-    Map<String, Object> extraClone = new HashMap<>();
+    Map<String, Object> extraClone = new ConcurrentHashMap<>();
```

Comment:
This is an important fix!

Reply:
not really IMO, Java guarantees that a field is thread-safe within a method, the global fields are using thread-safe map/list.
IMO this change is fine but has no effect

Reply:
Sorry but how does Java guarantee a field is thread safe?

This Map being an instance of `HashMap` and a thread going through it's keys while another is mutating it will throw on the reader. Or will it not?

Reply:
a field declared within a method is not accessible outside of it, so I don't see how `extraClone` could be modified by different threads.
`extra` yes, it could, but then we have a copy ref to avoid that

Reply:
> a field declared within a method is not accessible outside of it, so I don't see how extraClone could be modified by different threads.
extra yes, it could, but then we have a copy ref to avoid that

It won't be accessed while it's in this scope (as in this method's body and not accessible from the outside.
It will though be acessed by multiple threads once the Hub is global, with this Scope object global:

A few lines below:
`     clone.extra = extraClone; `

Now you can do `addExtra` via the public API and mutate this field.
Once someone wants to clone it next time, it'll iterate over this reference, which is not thread-safe